src/Nethermind/Nethermind.Db/LogIndexStorage.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Win32.SafeHandles;
+using Nethermind.Core;
+using Nethermind.Crypto;
+using Nethermind.Core.Crypto;
+using System.Linq;
+
+namespace Nethermind.Db
+{
+
+    public class LogIndexStorage
+    {
+        private Dictionary<AddressAsKey, int[]> _addressToBlocks = new Dictionary<AddressAsKey, int[]>();
+        private Dictionary<Hash256, int[]> _topicsToBlocks = new Dictionary<Hash256, int[]>();
+
+        public void LoadFile(Address address, string path)
```

Comment:
I Expect this is temporary?
We load the whole address index at once - we should load them page by page lazy.

Reply:
yup we were hoping we could test it out once we had the encoded data, but encoding it was taking too long